### PR TITLE
MW-310: Improve performance of RequisitionDto by utilizing search ord…

### PR DESCRIPTION
**Pull request by jkondrat**
…erables

---
This change is [Reviewable](https://reviewable.io/reviews/openlmis/openlmis-requisition/26)

### CODE REVIEW

File: src/main/java/org/openlmis/requisition/service/RequisitionService.java [hunk 1]
```diff
@@ -304,7 +304,9 @@ public Requisition reject(UUID requisitionId) {
           requisition.getSupervisoryNodeId(), userId);
 
       LOGGER.debug("Requisition rejected: {}", requisitionId);
-      requisition.reject(orderableReferenceDataService.findAll(), userId);
```
Comment (palbecki) [inline src/main/java/org/openlmis/requisition/service/RequisitionService.java]: why do we called findAll here?

Comment (jkondrat) [inline src/main/java/org/openlmis/requisition/service/RequisitionService.java]: Reject took a list of orderables in order to do some calculations on line items - each line item has corresponding orderable and we couldn't search them based on facility/program; I guess it was still better than calling findOne for each item as we did in ReqDtoBuilder

File: src/main/java/org/openlmis/utils/RequisitionExportHelper.java [hunk 2]
```diff
@@ -40,14 +42,19 @@
    */
   public List<RequisitionLineItemDto> exportToDtos(
```
Comment (palbecki) [inline src/main/java/org/openlmis/utils/RequisitionExportHelper.java]: please add some tests for that two methods if possible.

File: src/main/java/org/openlmis/requisition/web/RequisitionController.java [hunk 3]
```diff
@@ -261,7 +262,11 @@ public RequisitionDto submitRequisition(@PathVariable("id") UUID requisitionId)
     LOGGER.debug("Submitting a requisition with id " + requisition.getId());
 
     UserDto user = authenticationHelper.getCurrentUser();
-    requisition.submit(orderableReferenceDataService.findAll(), user.getId());
+
+    Set<UUID> orderableIds = requisition.getRequisitionLineItems().stream().map(
```
Comment (palbecki) [inline src/main/java/org/openlmis/requisition/web/RequisitionController.java]: Please  don't copy-paste these two lines.

File: src/main/java/org/openlmis/utils/RequisitionExportHelper.java (hunk 2 above)
Comment (jkondrat) [inline src/main/java/org/openlmis/utils/RequisitionExportHelper.java]: Done

File: src/main/java/org/openlmis/requisition/web/RequisitionController.java (hunk 3 above)
Comment (jkondrat) [inline src/main/java/org/openlmis/requisition/web/RequisitionController.java]: Ok, alright

**Pull request merged**